jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHandler.java

```diff
@@ -800,10 +823,7 @@ public void addListener(ListenerHolder listener)
     public void setListeners(ListenerHolder[] listeners)
     {
         if (listeners != null)
-            for (ListenerHolder holder : listeners)
-            {
-                holder.setServletHandler(this);
-            }
+            initializeHolders(listeners);
```

Comment:
So this means that in the past, adding a listener when the ServletHandler had already started would not have worked, because it never would have been started or initialized - unless someone _also_ happened to add a filter or servlet mapping.

Reply:
yep - 2nd bug!